### DISCUSSION TITLE
How to use layout with top-level routes

### DISCUSSION THREAD

**Discussion by JUNIORCO** [kiliman/remix-flat-routes#81] [Q&A] [answered]
Here's my `routes` layout:

```
.
└── routes/
    ├── auth+/
    │   ├── _layout.tsx
    │   ├── sign-in.tsx
    │   └── sign-up.tsx
    ├── _index.tsx
    ├── _layout.tsx
    ├── a.tsx
    └── b.tsx
```

`auth+/_layout.tsx` works, but I cant seem to get the root `_layout` to apply to `_index`, `a`, and `b`.

**Comment by kiliman** [comment, marked as answer]
Ah. The *_layout.tsx* trick only works inside a hybrid folder (ending in `+`). 

The *root* layout is the actual *app/root.tsx* file.

**Reply by JUNIORCO** [reply]
I see. And say I want to do

```
.
├── routes/
│   ├── auth+/
│   │   └── ...
│   └── app+/
│       ├── _layout.tsx
│       ├── a.tsx
│       └── b.tsx
└── root.tsx
```

Is there a way to have the `app` not show in the URL, so instead I may just access `/a` instead of `/app/a`? My guess is making the folder name `_app+`?

**Reply by kiliman** [reply]
Yes, use `_` prefix for pathless layouts (*routes/app+/_layout.tsx*).
